tobac/feature_detection.py

```diff
@@ -99,10 +128,42 @@ def feature_position(
         the first element is the feature position along the vertical dimension
         and the second two elements are the feature position on the first and
         second horizontal dimensions.
+        Note for PBCs: this point *can* be >hdim1_max or hdim2_max if the
+        point is between hdim1_max and hdim1_min. For example, if a feature
+        lies exactly between hdim1_max and hdim1_min, the output could be
+        between hdim1_max and hdim1_max+1. While a value between hdim1_min-1
+        and hdim1_min would also be valid, we choose to overflow on the max side of things.
     """
 
-    # are we 3D? if so, True.
-    is_3D = False
+    # First, if necessary, run PBC processing.
+    # processing of PBC indices
+    # checks to see if minimum and maximum values are present in dimensional array
+    # then if true, adds max value to any indices past the halfway point of their respective dimension.
+    # this, in essence, shifts the set of points to the high side.
+    pbc_options = ["hdim_1", "hdim_2", "both"]
+
+    if PBC_flag == "hdim_1" or PBC_flag == "both":
+        # periodic in y
+
+        if ((np.max(hdim1_indices)) == hdim1_max) and (
+            (np.min(hdim1_indices) == hdim1_min)
+        ):
+            # we have boundary points on each side of hdim1. Shift all points to the hdim1_max side.
+            for y2 in range(0, len(hdim1_indices)):
+                h1_ind = hdim1_indices[y2]
+                if h1_ind != hdim1_max:
+                    hdim1_indices[y2] = h1_ind + hdim1_max + 1
+
+    if PBC_flag == "hdim_2" or PBC_flag == "both":
+        # periodic in x
+
+        if ((np.max(hdim2_indices)) == hdim2_max) and (
+            (np.min(hdim2_indices) == hdim2_min)
+        ):
+            for x2 in range(0, len(hdim2_indices)):
+                h2_ind = hdim2_indices[x2]
+                if h2_ind != hdim2_max:
+                    hdim2_indices[x2] = h2_ind + hdim2_max + 1
```

Comment:
The changes here to fix `test_banded_feature` have lead to incorrect results for features that have points away from the boundaries. I've added a test for this to `test_feature_position_pbc`, and I'll see if we can use circmean for PBC's to handle all cases nicely

Reply:
Ok, so I've implemented a new method for finding the finding the centre/weighted average position of feature in a periodic domain. I've added some new tests for this, as well as added to the existing tests. I've updated `test_banded_feature` to modify the feature slightly, as before it was rotationally symmetric and so the PBC position was not well defined